src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -688,11 +680,51 @@ private void BackgroundLogger()
             finally
             {
                 _readyToCompress.Set();
-                if(_compressLoggingThread?.Join(1000) == false)
+                if (_compressLoggingThread?.Join(1000) == false)
                     _tokenSource.Cancel();
             }
         }
 
+        private void DisableLogsFor(List<WeakReference<LocalThreadWriterState>> threadStates, TimeSpan timeout)
+        {
+            var prevIsInfoEnabled = IsInfoEnabled;
+            var prevIsOperationsEnabled = IsOperationsEnabled;
+
+            try
+            {
+                IsInfoEnabled = false;
+                IsOperationsEnabled = false;
+
+                // discard all logs
+                foreach (var threadStateRef in threadStates)
+                {
+                    DiscardThreadLogState(threadStateRef);
+                }
+
+                foreach (var newThreadState in _newThreadStates)
+                {
+                    DiscardThreadLogState(newThreadState);
+                }
+
+                Thread.Sleep(timeout);
```

Comment:
Wouldn't the log fill up while we are sleeping?

Reply:
The logs are disabled at this point:
https://github.com/ravendb/ravendb/pull/9817/files#diff-cebe7d3f975916f1be4ab533d581c22aR695

Reply:
Okay